ui/pages/confirmations/components/confirm/info/shared/transaction-details/transaction-details.test.tsx

```diff
@@ -44,21 +44,96 @@ describe('<TransactionDetails />', () => {
     expect(container).toMatchSnapshot();
   });
 
-  it('renders component for transaction details with amount', () => {
-    const simulationDataMock = {
-      error: { code: SimulationErrorCode.Disabled },
-      tokenBalanceChanges: [],
-    };
-    const contractInteraction = genUnapprovedContractInteractionConfirmation({
-      simulationData: simulationDataMock,
-      chainId: CHAIN_IDS.GOERLI,
+  describe('AmountRow', () => {
+    describe('should be in the document', () => {
+      it('when showAdvancedDetails is true', () => {
+        const contractInteraction =
+          genUnapprovedContractInteractionConfirmation({
+            chainId: CHAIN_IDS.GOERLI,
+          });
+        const state = getMockConfirmStateForTransaction(contractInteraction, {
+          metamask: {
+            preferences: {
+              showConfirmationAdvancedDetails: true,
+            },
+          },
+        });
+        const mockStore = configureMockStore(middleware)(state);
+        const { getByTestId } = renderWithConfirmContextProvider(
+          <TransactionDetails />,
+          mockStore,
+        );
+        expect(
+          getByTestId('transaction-details-amount-row'),
+        ).toBeInTheDocument();
+      });
+
+      it('when value and simulated native balance mismatch', () => {
+        // Transaction value is set to 0x3782dace9d900000 below mock
+        const simulationDataMock = {
+          tokenBalanceChanges: [],
+          nativeBalanceChange: {
+            difference: '0x1' as Hex,
+            isDecrease: false,
+            previousBalance: '0x2' as Hex,
+            newBalance: '0x1' as Hex,
+          },
+        };
+        const contractInteraction =
+          genUnapprovedContractInteractionConfirmation({
+            simulationData: simulationDataMock,
+            chainId: CHAIN_IDS.GOERLI,
+          });
+        const state = getMockConfirmStateForTransaction(contractInteraction, {
+          metamask: {
+            preferences: {
+              // Intentionally setting to false to test the condition
+              showConfirmationAdvancedDetails: false,
+            },
+          },
+        });
+        const mockStore = configureMockStore(middleware)(state);
+        const { getByTestId } = renderWithConfirmContextProvider(
+          <TransactionDetails />,
+          mockStore,
+        );
+        expect(
+          getByTestId('transaction-details-amount-row'),
+        ).toBeInTheDocument();
+      });
+    });
+
+    it('should not be in the document when value and simulated native balance mismatch is within threshold', () => {
+      // Transaction value is set to 0x3782dace9d900000 below mock
+      const simulationDataMock = {
+        tokenBalanceChanges: [],
+        nativeBalanceChange: {
+          difference: '0x3782dace9d900000' as Hex,
```

Comment:
Can you use decimal test number constants and convert them to hex using a util, so the numbers are human readable?

Reply:
Make sense, done in https://github.com/MetaMask/metamask-extension/pull/29131/commits/9a6e27a6b81c4ddeee0f39733da32de10ea073bc